proposals/lip-0047.md

```diff
@@ -511,21 +647,29 @@ chainProperties.roundEndHeight = g.header.height + len(snapshotStore(0).addresse
 BFTThreshold = snapshotStore(0).threshold
 setBFTParameters(BFTThreshold, BFTThreshold, snapshotStore(0).validators)
```

Comment:
should't we abort the block execution if this function fails?

Reply:
Note that in this LIP we sometimes use `BFTThreshold` and sometimes `BFT_THRESHOLD`. As this is a variable and not a constant, I would suggest to switch to camelCase, e.g., `bftThreshold`.

Reply:
fixed variable names as jan suggested.
Regarding Andreas question, I think that is already addressed in line 644

Reply:
hm. I see 4 different cases where `[setBFTParameters](https://github.com/LiskHQ/lips/blob/c308e37920b31104da3f4ebd3eaebfdd8590a12f/proposals/lip-introduce_bft_module.md#execution-12)`, and already for the first I don't see clearly this will never happen. In particular, I do not see how line 644 prevents this.

Reply:
From my side the comment can be resolved.

Reply:
It does not prevent that if fails, what it says is that if `setBFTParameters` (for whatever the reason) fails, then the processing of this block halts and it is discarded.

It is up to the sidechain dev to set the right values for the BFT parameters so that this function does not fail.

Reply:
ok, I see that it is logically correct. But I think it's hard to get the logic from reading it. One reason is that it's not obvious at all that a setter function can fail. The second reason is now the inconsistency between calling `setBFTParameters` and calling `registerValidatorKeys`. For the latter one, the outcome of the call is explicitly used in the pseudo code. For the former one not. And it had to be done for `registerValidatorKeys` because "failing" is not well defined for this function.